chore: rebalance embedded login messaging and add hybrid-use framing (#1327)
auth0/docs-v2: main/docs/authenticate/login/cross-origin-authentication.mdx

description: An explanation of cross-origin authentication in Auth0 and its compatibility with browsers
title: Cross-Origin Authentication
---

> Cross-origin authentication is the flow Auth0 uses when an embedded login form (the Lock widget or a custom login form) collects credentials in your application and submits them to an Auth0 domain. Auth0 provides this flow with built-in protections—including third-party cookies for browser-side checks—so authentication can be performed securely across origins.

Reviewer: ```suggestion
Auth0 uses the cross-origin authentication flow when an embedded login form (the Lock widget or a custom login form) collects credentials in your application and submits them to an Auth0 domain. On Auth0, this flow has built-in protections, including third-party cookies for browser-side checks, for better authentication security across origins.
```

> The use of third-party cookies allows Lock and Auth0's backend to perform the necessary checks to prevent phishing and other cross-origin attacks when creating a <Tooltip tip="Single Sign-On (SSO): Service that, after a user logs into one applicaton, automatically logs that user in to other applications." cta="View Glossary" href="/docs/glossary?term=Single+Sign-on">Single Sign-on</Tooltip> experience with the Lock widget or a custom login form, and it also produces a secure login experience even if SSO is not a goal.

Reviewer: ```suggestion
Using third-party cookies allows Lock and Auth0's backend to perform the necessary checks to prevent phishing and other cross-origin attacks when creating a <Tooltip tip="Single Sign-On (SSO): Service that, after a user logs into one applicaton, automatically logs that user in to other applications." cta="View Glossary" href="/docs/glossary?term=Single+Sign-on">single sign-on</Tooltip> (SSO) experience with the Lock widget or a custom login form. It also produces a secure login experience even if SSO is not a goal.
```

## Browser considerations

> Modern browsers (including Firefox, Safari with ITP, and Chromium-based browsers) restrict or block third-party cookies by default for privacy reasons. Web applications relying on third-party cookies for cross-origin authentication may fail in those browsers. To make embedded login work reliably across browsers, configure your tenant so the application and Auth0 share a top-level domain. This is also considered a security best practice.

Reviewer 2: The best security practice might be confusing here. Maybe we can edit the last sentence to be something like:
Also, as a security best practice, we recommend using a [custom domain](/docs/customize/custom-domains) or multiple custom domains with the same top-level domain.

Author: I've updated the wording here to hopefully be clearer

To make embedded login work across all browsers:

* **(Recommended) Enable a [Custom Domain](/docs/customize/custom-domains)** on your tenant and host your web application on the same top-level domain as your Auth0 custom domain. For example, host your app at `https://northwind.com` and your Auth0 custom domain at `https://login.northwind.com`. The cookies are no longer third-party because both your Auth0 tenant and your application share the same top-level domain, so browsers do not block them. This is the recommended path for production embedded web applications.

> * **Create and link a cross-origin verification page** that allows cross-origin authentication to work in a limited number of browsers when third-party cookies are still permitted. This is suitable for testing or controlled environments where browser coverage requirements are well understood.

Reviewer: ```suggestion
Modern browsers (including Firefox, Safari with ITP, and Chromium-based browsers) restrict or block third-party cookies by default for privacy reasons. Web applications relying on third-party cookies for cross-origin authentication may fail in those browsers.

To make embedded login work reliably across browsers for production embedded web applications, use the same top-level domain for your application and Auth0 tenant. For example, host your app at `https://example.com` and your Auth0 custom domain at `https://login.example.com`.

With this configuration, the cookies are no longer third party and browsers do not block them. As a security best practice, we recommend using a [custom domain](/docs/customize/custom-domains) or multiple custom domains with the same top-level domain. 
 
For testing or in controlled environment with clear browser coverage requirements, you can create and link a cross-origin verification page that allows cross-origin authentication to work in those specific browsers when third-party cookies are still permitted.
```
